### PR TITLE
spine: Fix the "mark all complete" checkbox

### LINKED ISSUE #795: App does not update the state of the 'mark all complete' checkbox
The 'mark all complete' checkbox should be cleared if the all the todo items are checked then the user un-checks an item. i.e. the UI state should remain consistent.

This affects the following:

 - [ ] spine
 - [x] angularjs-perf
 - [x] kendo

See the following test failure:

    $ mocha allTests.js  -R list --framework=spine

      ․ TodoMVC - spine (1/1) No Todos should hide #main and #footer: 215ms
      ․ TodoMVC - spine (1/1) New Todo should allow me to add todo items: 553ms
      ․ TodoMVC - spine (1/1) New Todo should clear text input field when an item is added: 287ms
      ․ TodoMVC - spine (1/1) New Todo should trim text input: 321ms
      ․ TodoMVC - spine (1/1) New Todo should show #main and #footer when items added: 405ms
      ․ TodoMVC - spine (1/1) Mark all as completed should allow me to mark all items as completed: 933ms
      ․ TodoMVC - spine (1/1) Mark all as completed should allow me to clear the completion state of all items: 1025ms
      1) TodoMVC - spine (1/1) Mark all as completed complete all checkbox should update state when items are completed / cleared
      ․ TodoMVC - spine (1/1) Item should allow me to mark items as complete: 868ms
      ․ TodoMVC - spine (1/1) Item should allow me to un-mark items as complete: 826ms
      ․ TodoMVC - spine (1/1) Item should allow me to edit an item: 1098ms
      ․ TodoMVC - spine (1/1) Item should show the remove button on hover: 10ms
      ․ TodoMVC - spine (1/1) Editing should hide other controls when editing: 715ms
      ․ TodoMVC - spine (1/1) Editing should save edits on enter: 1079ms
      ․ TodoMVC - spine (1/1) Editing should save edits on blur: 1168ms
      ․ TodoMVC - spine (1/1) Editing should trim entered text: 1131ms
      ․ TodoMVC - spine (1/1) Editing should remove the item if an empty text string was entered: 1013ms
      2) TodoMVC - spine (1/1) Editing should cancel edits on escape
      ․ TodoMVC - spine (1/1) Counter should display the current number of todo items: 426ms
      ․ TodoMVC - spine (1/1) Clear completed button should display the number of completed items: 870ms
      ․ TodoMVC - spine (1/1) Clear completed button should remove completed items when clicked: 844ms
      ․ TodoMVC - spine (1/1) Clear completed button should be hidden when there are no items that are completed: 867ms
      ․ TodoMVC - spine (1/1) Persistence should persist its data: 2652ms
      ․ TodoMVC - spine (1/1) Routing should allow me to display active items: 861ms
      ․ TodoMVC - spine (1/1) Routing should allow me to display completed items: 815ms
      ․ TodoMVC - spine (1/1) Routing should allow me to display all items: 1134ms
      ․ TodoMVC - spine (1/1) Routing should highlight the currently applied filter: 961ms

      25 passing (1m)
      2 failing

### PR DESCRIPTION
**Pull request by dmethvin**
Fixes #795

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -34,6 +34,7 @@
 		"examples/humble/js/**",
 		"examples/js_of_ocaml/js/*.js",
 		"examples/polymer/elements/elements.build.js",
+		"examples/spine/js/**",
```
Comment (dmethvin) [inline .jscsrc]: Excluding the js files since the source is Coffeescript.

File: examples/spine/src/app.coffee [hunk 2]
```diff
@@ -57,23 +57,25 @@ class TodoApp extends Spine.Controller
 		@addNew todo for todo in @getByFilter()
 
 	toggleAll: (e) ->
+		checked = e.target.checked
```
Comment (dmethvin) [inline examples/spine/src/app.coffee]: The "mark all complete" checkbox can change during the loop below so the initial value needs to be cached.

**Comment by samccone** [comment]
@dmethvin mind rebasing off of master?

**Comment by dmethvin** [comment]
@samccone Updated, but it looks like some router-related tests have been added that this doesn't pass? I didn't have enough stamina to get the tests running here locally. The readme instructions in the /test folder seem to be incomplete, at least for Windows, because it can't find the Chrome driver.
capture

File: examples/spine/src/app.coffee [hunk 3]
```diff
@@ -57,23 +57,25 @@ class TodoApp extends Spine.Controller
...
 	clearCompletedItem: ->
 		Todo.destroyCompleted()
 
 	toggleElems: =>
-		isTodos = !!Todo.count()
-		@main.toggle isTodos
-		@footer.toggle isTodos
-		@clearCompleted.toggle !!Todo.completed().length
-		@toggleAllElem.removeAttr 'checked' if !Todo.completed().length
+		dones = Todo.completed().length
```
Comment (samccone) [inline examples/spine/src/app.coffee]: as much as I like the word dones :) perhaps completed might be nicer

:goat: :pizza:

Comment (dmethvin) [inline examples/spine/src/app.coffee]: Well it certainly will gzip better, given all the completeds in the vicinity. :smile:

**Comment by dmethvin** [comment]
I got the unit tests to run locally by manually downloading Chromedriver for Windows. :star2:

It looks like the remaining test that fails is already noted in #858 so this PR at least is ready to land.

**Comment by samccone** [comment]
I like this work

good work :dancer:

**Pull request merged**